pkg/strategy/controller.go

```diff
@@ -135,23 +135,48 @@ func normalizeRollover(spec *v1.JaegerEsRolloverSpec) {
 }
 
 func normalizeUI(spec *v1.JaegerSpec) {
-	sOpts := spec.Storage.Options.Map()
 	uiOpts := map[string]interface{}{}
 	if !spec.UI.Options.IsEmpty() {
 		if m, err := spec.UI.Options.GetMap(); err == nil {
 			uiOpts = m
 		}
 	}
-	// we respect explicit UI config
-	if _, ok := uiOpts["archiveEnabled"]; ok {
+	enableArchiveButton(uiOpts, spec.Storage.Options.Map())
+	disableDependenciesTab(uiOpts, spec.Storage.Type, spec.Storage.SparkDependencies.Enabled)
+	if len(uiOpts) > 0 {
+		spec.UI.Options = v1.NewFreeForm(uiOpts)
+	}
+}
+
+func enableArchiveButton(uiOpts map[string]interface{}, sOpts map[string]string) {
+	// respect explicit settings
+	if _, ok := uiOpts["archiveEnabled"]; !ok {
+		// archive tab is by default disabled
+		if strings.EqualFold(sOpts["es-archive.enabled"], "true") ||
+			strings.EqualFold(sOpts["cassandra-archive.enabled"], "true") {
+			uiOpts["archiveEnabled"] = true
+		}
+	}
+}
+
+func disableDependenciesTab(uiOpts map[string]interface{}, storage string, depsEnabled *bool) {
+	// dependency tab is by default enabled and memory storage support it
+	if strings.EqualFold(storage, "memory") || (depsEnabled != nil && *depsEnabled == false) {
```

Comment:
Just realised, with the additional test, if dependencies are explicitly disabled, then menuEnabled should be false?

Reply:
so should the condition be `if strings.EqualFold(storage, "memory") || depsEnabled == nil || *depsEnabled == true) {`?

Reply:
You are right, changing